assets/src/dashboard/app/api/useStoryApi.js

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useReducer } from 'react';
+import moment from 'moment';
+import queryString from 'query-string';
+
+/**
+ * Internal dependencies
+ */
+import {
+  STORY_STATUSES,
+  STORY_SORT_OPTIONS,
+  ORDER_BY_SORT,
+  ITEMS_PER_PAGE,
+} from '../../constants';
+import storyReducer, {
+  defaultStoriesState,
+  ACTION_TYPES as STORY_ACTION_TYPES,
+} from '../reducer/stories';
+
+export function reshapeStoryObject(editStoryURL) {
+  return function ({ id, title, modified, status, story_data: storyData }) {
+    if (
+      !Array.isArray(storyData.pages) ||
+      !id ||
+      storyData.pages.length === 0
+    ) {
+      return null;
+    }
+    return {
+      id,
+      status,
+      title: title.rendered,
+      modified: moment(modified),
+      pages: storyData.pages,
+      centerTargetAction: '',
+      bottomTargetAction: `${editStoryURL}&post=${id}`,
+    };
+  };
+}
+
+const useStoryApi = (dataAdapter, { editStoryURL, wpApi }) => {
+  const [state, dispatch] = useReducer(storyReducer, defaultStoriesState);
+
+  const fetchStories = useCallback(
+    async ({
+      status = STORY_STATUSES[0].value,
+      sortOption = STORY_SORT_OPTIONS.LAST_MODIFIED,
+      sortDirection,
+      searchTerm,
+      page = 1,
+      perPage = ITEMS_PER_PAGE,
+    }) => {
+      dispatch({
+        type: STORY_ACTION_TYPES.LOADING_STORIES,
+        payload: true,
+      });
+
+      if (!wpApi) {
+        dispatch({
+          type: STORY_ACTION_TYPES.FETCH_STORIES_FAILURE,
+          payload: true,
+        });
+        return [];
```

Comment:
Do we still need to return anything since this all is dispatching to the reducer? I think we can just `return` without anything.